packages/contracts/src/EmailWalletCore.sol

```diff
@@ -549,19 +579,29 @@ contract EmailWalletCore is ReentrancyGuard, OwnableUpgradeable, UUPSUpgradeable
         require(state.length > 0, "state cannot be empty");
         require(extensionAddr != address(0), "invalid extension contract");
         require(emailAddrCommit != bytes32(0), "invalid emailAddrCommit");
+        // Sora's comment: the following check is performed in `validateEmailOp`, right?
+        // require(unclaimedStateOfEmailAddrCommit[emailAddrCommit].sender == address(0), "unclaimed state exists");
 
         uint256 expiryTime = block.timestamp + unclaimedFundExpiryDuration;
 
         UnclaimedState memory us = UnclaimedState({
             emailAddrCommit: emailAddrCommit,
             extensionAddr: extensionAddr,
             sender: currContext.walletAddr,
-            state: state
+            state: state,
+            expiryTime: expiryTime
         });
 
+        // Sora's comment: `extension.registerUnclaimedState` should be called here.
+        // Sora's comment: Here try-and-catch should be used to prevent a malicious extension that returns error.  
+        // The max gas to call it is calculated from `unclaimedStateRegisterGas`.
+        Extension extension = Extension(extensionAddr);
+        bool registered = extension.registerUnclaimedState(us,true);
+        require(registered, "unclaimed state reg failed");
+
```

Comment:
Extensions can handle necessary operations before registering an unclaimed state internally.
However, I think it is easier to understand for developers that `registerUnclaimedState ` is always called when the unclaimed state is registered, and it can know if the unclaimed state is registered internally or not.